src/AutoRest.CSharp/Output/Models/RestClient.cs

```diff
@@ -166,103 +166,214 @@ private string GetRequestParameterName (RequestParameter requestParameter)
 
         private RestClientMethod BuildMethod(Operation operation, HttpRequest httpRequest, ICollection<RequestParameter> requestParameters, ResponseHeaderGroupType? responseHeaderModel)
         {
-            HttpWithBodyRequest? httpRequestWithBody = httpRequest as HttpWithBodyRequest;
-            Dictionary<string, PathSegment> uriParameters = new Dictionary<string, PathSegment>();
-            Dictionary<string, PathSegment> pathParameters = new Dictionary<string, PathSegment>();
-            List<QueryParameter> query = new List<QueryParameter>();
-            List<RequestHeader> headers = new List<RequestHeader>();
-            Dictionary<RequestParameter, ReferenceOrConstant> allParameters = new Dictionary<RequestParameter, ReferenceOrConstant>();
-            Dictionary<RequestParameter, Parameter> methodParameters = new Dictionary<RequestParameter, Parameter>();
+            Dictionary<RequestParameter, ConstructedParameter> allParameters = new ();
```

Comment:
Random style question. Why this and not:

`var allParameters = new Dictionary<RequestParameter, ConstructedParameter> ();`

I don't think I've seen new () used much before, just a preference or something deeper?

Reply:
Feel free to override me on this. I don't have a strong opinion

We have a rule where you are required to specify the variable type when it's not obvious.

When using var it can result in:

```C#
var v1 = new List<T1>();
List<T2> v2 = GetList();
```

When using `new()` the resulting code is more uniform:

```C#
List<T1> v1 = new ();
List<T2> v2 = GetList();
```

Reply:
No strong opinion either way. Makes sense though.